crates/engine/tree/src/tree/mod.rs

```diff
@@ -2188,16 +2189,31 @@ where
         let block = block.unseal();
 
         let exec_time = Instant::now();
-        // TODO: create StateRootTask with the receiving end of a channel and
-        // pass the sending end of the channel to the state hook.
-        let noop_state_hook = |_result_and_state: &ResultAndState| {};
+
+        let (state_root_tx, state_root_rx) = unbounded_channel();
+        // TODO: create consistent_view using `new_with_latest_tip` when we
+        // switch to `StateRootTask`.
+        let consistent_view = ConsistentDbView::new(self.provider.clone(), None);
+        let state_root_task = StateRootTask::new(
+            consistent_view,
+            // TODO: calculate `TrieInput` like in `self.compute_state_root_parallel`.
+            Arc::new(TrieInput::default()),
+            UnboundedReceiverStream::new(state_root_rx),
+        );
+        let state_root_handle = state_root_task.spawn();
+        let state_hook = move |result_and_state: &ResultAndState| {
+            let _ = state_root_tx.send(result_and_state.state.clone());
+        };
```

Comment:
let's leave the integration for once we finish with the task? i'd prefer to have it ready before enabling it in the tree

Reply:
makes total sense, done here https://github.com/paradigmxyz/reth/pull/12335/commits/ba4a80401b0108514939be04161f7ad4e113b459 i'll add some unit tests on a follow up to cover the existing functionality